rest_framework_json_api/schemas/openapi.py

```diff
@@ -0,0 +1,839 @@
+        :param view_endpoints: list of all view endpoints
+        :param related_serializer: the related serializer for a given related field
+        :param parent_view: the parent view (used to find toMany vs. toOne).
+        :return:view
+        """
+        for path, method, view in view_endpoints:
+            view_serializer = view.get_serializer()
+            if not isinstance(related_serializer, type):
+                related_serializer_class = import_class_from_dotted_path(related_serializer)
+            else:
+                related_serializer_class = related_serializer
+            if isinstance(view_serializer, related_serializer_class):
+                return view
+
+        return None
+
+    def _field_is_one_or_many(self, field, view):
+        serializer = view.get_serializer()
+        if isinstance(serializer.fields[field], ManyRelatedField):
+            return 'list'
+        else:
+            return 'fetch'
+
+
+class AutoSchema(drf_openapi.AutoSchema):
+    """
+    Extend DRF's openapi.AutoSchema for JSONAPI serialization.
+    """
+    content_types = ['application/vnd.api+json']
+
+    def __init__(self):
+        """
+        Initialize the JSONAPI OAS schema generator
+        """
+        super().__init__()
+
+    def get_operation(self, path, method, action=None):
+        """ basically a copy of AutoSchema.get_operation """
+        operation = {}
+        operation['operationId'] = self._get_operation_id(path, method)
+        operation['description'] = self.get_description(path, method)
+
+        parameters = []
+        parameters += self._get_path_parameters(path, method)
+        # pagination, filters only apply to GET/HEAD of collections and items
+        if method in ['GET', 'HEAD']:
+            parameters += self._get_include_parameters(path, method)
+            parameters += self._get_fields_parameters(path, method)
+            parameters += self._get_sort_parameters(path, method)
+            parameters += self._get_pagination_parameters(path, method)
+            parameters += self._get_filter_parameters(path, method)
+        operation['parameters'] = parameters
+
+        # get request and response code schemas
+        if method == 'GET':
+            if is_list_view(path, method, self.view):
+                self._get_collection(operation)
+            else:
+                self._get_item(operation)
+        elif method == 'POST':
+            self._post_item(operation, path, action)
+        elif method == 'PATCH':
+            self._patch_item(operation, path, action)
+        elif method == 'DELETE':
+            # should only allow deleting a resource, not a collection
+            # TODO: delete of a relationship is different.
+            self._delete_item(operation, path, action)
+        return operation
+
+    def _get_operation_id(self, path, method):
+        """ create a unique operationId """
```

Comment:
Is this still the case that this needs to be overwritten? I don't quite see why this should be different in DJA then in DRF.

Reply:
DRF was coming up with non-unique operation_ids and I think it still does that.

Also note that drf has renamed many of the private methods to public (dropping the leading `_`) so there's some hope of stability:-)

Reply:
DRF is still coming up with non-unique operation_ids due to the multiple paths for a given view. For example:
```yaml
/v1/courses/{id}/:
/v1/courses/{id}/relationships/course_terms/:
/v1/courses/{id}/course_terms/:
```
So this overrides by using the path.